include/real/exact_number.hpp

```diff
@@ -1222,8 +1214,11 @@ namespace boost {
                 if (exponent < 0) {
                     return false;
                 } else {
-                    /// @TODO: ensure it doesn't overflow
-                    return (exponent_t) digits.size() <= exponent;
+                    if (exponent < 0)
+                        return false;
+                    else {
+                        return digits.size() <= (size_t) exponent;
```

Comment:
This is in the territory of undefined behavior:

Example:
```
    int i = -1;
    std::size_t s = (std::size_t) i;
    std::cout << s << std::endl;
```

outputs: 18446744073709551615

Reply:
That is why I return false if exponent < 0. At that point, exponent is always >= 0 so it should not have problems when cast

Reply:
But why you check twice?
```
if (exponent < 0) {
                    return false;
                } else {
                    if (exponent < 0)
                        return false;
...
```

It seems like the second if is deadcode

Reply:
Oops. I'll remove the redundancy.